File: runtime/go/units/wasi/fd_advise.go
// fd_advise is purely advisory: validate the advice code and return success.
// A directory fd (no *os.File) is BADF, matching the suite's accepted
// ISDIR/BADF/NOTCAPABLE for directory ops.
func (w *WASI) wasi_fd_advise(fd uint32, offset, length uint64, advice uint32) uint32 {
    if _, ok := w.fds[fd].(*os.File); !ok {
        return wasiBadf
    }
    if advice > 5 { // NORMAL, SEQUENTIAL, RANDOM, WILLNEED, DONTNEED, NOREUSE
        return wasiInval
    }
    return wasiOk
}

File: crates/dewasm-backend-java/units/wasi/fd_advise.java
// Access-pattern advice is advisory only: this runtime keeps no page cache to tune, so a valid request is accepted as a no-op.
// The fd must be a regular file and the advice one of the six WASI codes (NORMAL..NOREUSE).
int wasi_fd_advise(int fd, long offset, long len, int advice) {
    Object e = fds.get(fd);
    if (!(e instanceof Handle)) {
        return WASI_BADF;
    }
    if (advice < 0 || advice > 5) {
        return WASI_INVAL;
    }
    return WASI_OK;
}
